src/main/java/org/craftercms/deployer/impl/processors/FileOutputProcessor.java

```diff
@@ -73,6 +73,10 @@ public void destroy() throws DeployerException {
 
     @Override
     protected void doExecute(Deployment deployment) throws DeployerException {
+        if(deployment.isChangeSetEmpty()) {
```

Comment:
See the lines below where it checks if the there are created, updated or deleted files. If there aren't any, it returns an empty list BUT still prints the record. Please do the same if the changeSet is null. Another option is to make sure the deployment creates a changeSet that is completely empty instead of null, before the processors are executed.